solr/core/src/java/org/apache/solr/query/FilterQuery.java

```diff
@@ -89,8 +90,10 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
       return new ConstantScoreQuery(q).createWeight(searcher, scoreMode, 1f);
     }
 
+    SolrIndexSearcher solrSearcher = (SolrIndexSearcher)searcher;
+    DocSet docs = solrSearcher.getDocSet(q);
     // reqInfo.addCloseHook(docs);  // needed for off-heap refcounting
 
-    return new ConstantScoreQuery(q).createWeight(searcher, scoreMode, 1f);
+    return new ConstantScoreQuery(docs.makeQuery()).createWeight(searcher, scoreMode, 1f);
```

Comment:
DocSet.makeQuery *is* a constant scoring query, so wrapping in ConstantScoreQuery is pointless.

Reply:
I will fix that, thanks